pilz_robot_tfc_api/src/tfc_api.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) 2019 Pilz GmbH & Co. KG
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""API for easy usage of Pilz robot test facility."""
+
+OP_MODE_T1 = 0
+OP_MODE_T2 = 1
+OP_MODE_AUTO = 2
+
+
+class TestFacilityControlAPI(Object):
+    """ API for controling the PILZ robot test facility """
+      
+    
+    def chooseOperationMode(self, op_mode):
+        """ Allows to change the operation mode """
+        raise NotImplementedError
+
+    def activateEmergency(self):
+        """ Triggers emergency stop by releases the STO_A und STO_B (STO-Safe Torque Off)"""
+        raise NotImplsementedError
+
+    def disableEmergency(self):
+        """ Realeses emergency stop by triggering the STO_A und STO_B (STO-Safe Torque Off) """
+        raise NotImplementedError
+
+    def acknowledgeReadySignal(self):
+        """ Acknowledges the rady signal of the safety controller """
+        raise NotImplementedError
+
+    def activateEnabling(self):
+        """ Enables the drives by triggering STO_A und STO_B (STO-Safe Torque Off) """
+        raise NotImplementedError
+
+    def disableEnabling(self):
```

Comment:
I do not know, what has been discussed, but I do not like the name `disableEnabling`.
Possible alternatives: deactivateEnabling? or simply enable/disable?

Reply:
The docus for `activateEnabling` and `disableEmergency` are identical. So one of the functions is redundant. Or the docu is not sufficient.

Reply:
> I do not know, what has been discussed, but I do not like the name `disableEnabling`.
> Possible alternatives: deactivateEnabling? or simply enable/disable?

Changed to "deactive_enabling"

Reply:
> The docus for `activateEnabling` and `disableEmergency` are identical. So one of the functions is redundant. Or the docu is not sufficient.

Yeah, that's a tricky one. The problem is: We want to "emulate" the real hardware setup. The real hardware setup consists of a enabling button and an emergency stop. Both do (if I'm correct) basically the same, hence, the same docu. However, for a correct test we need both functionalities. Therefore, I opt to leave both functions.

Reply:
Yes, I understand, that it makes sense to keep both. But:
```
by triggering STO_A und STO_B
```
I think this is too specific for the general API. From this point of view it is not necessary that both functions do the same. This just follows from the actual implementation.

Reply:
Done. -> Improved API documentation.